src/components/network-area-diagram-viewer/network-area-diagram-viewer.ts

```diff
@@ -362,6 +374,40 @@ export class NetworkAreaDiagramViewer {
             }
         }
     }
+    private onHover(mouseEvent: MouseEvent) {
+        const equipmentsWithPopover = ['LINE', 'TWO_WINDINGS_TRANSFORMER', 'PHASE_SHIFT_TRANSFORMER'];
+        if (this.handleTogglePopover == null) {
+            return;
+        }
+
+        const hoverableElem = DiagramUtils.getHoverableForm(mouseEvent.target as SVGElement);
+        const parentElement = hoverableElem?.parentElement;
+        if (hoverableElem == null || parentElement == null) {
+            this.handleTogglePopover(false, null, '', '');
+            return;
+        }
+        if (!DiagramUtils.classIsContainerOfHoverables(parentElement)) {
+            this.handleTogglePopover(false, null, '', '');
+            return;
+        }
```

Comment:
Couldn't his be simplifified with a single `if (!hoverableElem)`?

Besides, if `!hoverableElem` then it means we're hovering over something of no interest. If I understand properly your need, with the shouldDisplay boolean, and the "toggle" naming, you want to be notified when you just stopped hovering a line for instance. Is that the usecase? I'm not sure to get the sense of shouldDisplay in fact. And I don't see the point to call the callback with no equipment id provided.

Reply:
with the logic I provided above for `getHoverableFrom`, hoverableElem is null if it has no parent element.

Ok, you need to know when the hovering stopped, I understand. It would make more sense to give the previous equipmentId and type don't you think? They're probably not needed, but it makes more sense to say what is not anymore hovered.

If you need information, we should probably handle the case of hovering a line and then going directly to another one. You can do that if the lines are crossing. We should probably call twice the callback, first with false and the old id, then with true and the new id?